### PR TITLE
Fix : Responsive conditions appear at the inappropriate position in other blocks

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://app.startinfinity.com/b/Mi5zu3VRVzk/M7Dpq9yR4Pe/999e91fd-4a41-4ad0-ab8e-0e33a57b6554?redirect=true&view=6d3bcc3d-b047-4739-8e05-d9621e9f9445&t=attributes

### Screenshots

- https://share.getcloudapp.com/7KuQPArP

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Drag and drop core/gallery block add an image.
- And check our extension tab should be visible at last above the advanced tab.
- Check with other blocks too.
- For Responsive condition - Show which block is hidden in the editor the same as elementor. ( Test cases added in card )
- [Screencast](https://share.getcloudapp.com/p9u6k62q) (different condition to diff block)

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/extensions/condition-block/index.js [hunk 1]
```diff
@@ -169,6 +171,32 @@ const AdvancedControlsBlock = createHigherOrderComponent( ( BlockEdit ) => {
...
 		const customBlocks = uagb_blocks_info.uagb_enable_extensions_for_blocks;
 		const blockPrefix = blockName.substring( 0, blockName.indexOf( '/' ) + 1 );
+		const { getSelectedBlock } = select( 'core/block-editor' );
+		let childBlocks = [];
+		if ( getSelectedBlock()?.innerBlocks ) {
+			childBlocks = getSelectedBlock().innerBlocks;
+		}
+		useEffect( () => {
+
+				let DisplayPanelBody = document.querySelector( '.uag-advance-panel-body-display' );
```
Comment (RajkiranBagal) [inline src/blocks/extensions/condition-block/index.js:181]: use camelCases for JS variables

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/extensions/condition-block/index.js (hunk 1 above)
Comment (AbhijitNage123) [reply src/blocks/extensions/condition-block/index.js]: Done

File: src/blocks/extensions/condition-block/index.js [hunk 2]
```diff
@@ -169,6 +171,32 @@ const AdvancedControlsBlock = createHigherOrderComponent( ( BlockEdit ) => {
 		const excludeBlocks = uagb_blocks_info.uagb_exclude_blocks_from_extension;
 		const customBlocks = uagb_blocks_info.uagb_enable_extensions_for_blocks;
 		const blockPrefix = blockName.substring( 0, blockName.indexOf( '/' ) + 1 );
+		const { getSelectedBlock } = select( 'core/block-editor' );
+		let childBlocks = [];
+		if ( getSelectedBlock()?.innerBlocks ) {
+			childBlocks = getSelectedBlock().innerBlocks;
```
Comment (RajkiranBagal) [inline src/blocks/extensions/condition-block/index.js:177]: line 175 -> 178 can be converted to:
`let childBlocks = getSelectedBlock()?.innerBlocks;`

File: src/blocks/extensions/condition-block/index.js [hunk 3]
```diff
@@ -169,6 +171,32 @@ const AdvancedControlsBlock = createHigherOrderComponent( ( BlockEdit ) => {
...
+		}
+		useEffect( () => {
+
+				let displayPanel = document.querySelector( '.uag-advance-panel-body-display' );
+				let responsivePanel = document.querySelector( '.uag-advance-panel-body-responsive' );
+				let masonryPanel = document.querySelector( '.uag-advance-panel-body-masonry' );
+				if( displayPanel ){
+					const tabsParent = displayPanel.parentElement;
+					if( tabsParent ){
+						displayPanel = displayPanel ? displayPanel : '';
```
Comment (RajkiranBagal) [inline src/blocks/extensions/condition-block/index.js:187]: displayPanel is already checked above, so this condition is meaningless

File: src/blocks/extensions/condition-block/index.js [hunk 4]
```diff
@@ -169,6 +171,32 @@ const AdvancedControlsBlock = createHigherOrderComponent( ( BlockEdit ) => {
...
+						responsivePanel = responsivePanel ? responsivePanel : '';
+						masonryPanel = masonryPanel ? masonryPanel : '';
+						if( tabsParent.lastChild.className.includes( 'components-panel__body is-opened' ) ){
```
Comment (RajkiranBagal) [inline src/blocks/extensions/condition-block/index.js:190]: Why this condition is needed?

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/extensions/condition-block/index.js (hunk 4 above)
Comment (AbhijitNage123) [reply src/blocks/extensions/condition-block/index.js:186]: @RajkiranBagal tabsParent returning a masonry panel body so we need to check if there any core `'components-panel__body is-opened'` is present or not. See the screencast here - https://share.getcloudapp.com/d5u9Anv7

File: src/blocks/extensions/condition-block/index.js (hunk 3 above)
Comment (AbhijitNage123) [reply src/blocks/extensions/condition-block/index.js]: Resolved.

File: src/blocks/extensions/condition-block/index.js (hunk 2 above)
Comment (AbhijitNage123) [reply src/blocks/extensions/condition-block/index.js]: Resolved.

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**